presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java

```diff
@@ -13,40 +13,46 @@
  */
 package com.teradata.prestomanager.controller;
 
+import com.google.common.collect.ImmutableSet;
 import com.google.inject.Inject;
+import io.airlift.discovery.client.DiscoveryException;
+import io.airlift.discovery.client.ServiceDescriptor;
+import io.airlift.discovery.client.ServiceSelector;
+import io.airlift.discovery.client.ServiceType;
+import io.airlift.log.Logger;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Collection;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.UUID;
-import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
 public class NodeSet
 {
-    private Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+    private static final Logger LOG = Logger.get(NodeSet.class);
```

Comment:
I think it's fine either way.

(I prefer `LOG` because it's shorter, and then you can read "LOG.error" as "log error".)